### PR TITLE
Seperate mk1 and mk2 deployment

### PR DESCRIPTION
**Pull request by Furisto**
## Description
Seperates mk1 and mk2 deployment. Either mk1 or mk2 will be deployed but not both.

## Related Issue(s)

Fixes WKS-25

## How to test
- Create ephemeral cluster with https://github.com/gitpod-io/ops/tree/fo/single-manager and https://github.com/gitpod-io/ops/tree/fo/single-manager-mk2
- Run an image build

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

**Review by aledbf — APPROVED** [review]

**Comment by kylos101** [comment]
@Furisto all PRs have a hold on them now by default :information_source: , please remove the hold when ready.

File: install/installer/pkg/components/image-builder-mk3/deployment.go [hunk 1]
```diff
@@ -60,6 +62,16 @@ func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
 		return nil, err
 	}
 
+	wsmanSecret := wsmanager.TLSSecretNameClient
+	_ = ctx.WithExperimental(func(ucfg *experimental.Config) error {
+		if ucfg.Workspace != nil && ucfg.Workspace.UseWsmanagerMk2 {
+			wsmanSecret = wsmanagermk2.TLSSecretNameClient
+		} else {
```
[the remark this answers is not in the document]
Comment (Furisto) [reply install/installer/pkg/components/image-builder-mk3/deployment.go:69]: @aledbf I do not understand what you mean by that. Can you expand?

[the remark this answers is not in the document]
Comment (aledbf) [reply install/installer/pkg/components/image-builder-mk3/deployment.go:69]: Sorry, the else is not necessary. You defined wsmanSecret using the value from the else section

[the remark this answers is not in the document]
Comment (Furisto) [reply install/installer/pkg/components/image-builder-mk3/deployment.go]: :facepalm: :+1:

**Pull request merged**